draft-hendrickson-privacypass-public-metadata-issuance.md

```diff
@@ -43,10 +46,220 @@ TODO Abstract
+TokenResponse structure:
+
+~~~
+struct {
+  uint8_t blind_sig[Nk];
+} TokenResponse;
+~~~
+
+The Issuer generates an HTTP response with status code 200 whose content
+consists of TokenResponse, with the content type set as
+"application/private-token-response".
+
+~~~
+:status = 200
+content-type = application/private-token-response
+content-length = <Length of TokenResponse>
+
+<Bytes containing the TokenResponse>
+~~~
+
+## Finalization {#public-finalize}
+
+Upon receipt, the Client handles the response and, if successful, processes the
+content as follows:
+
+~~~
+authenticator =
+  Finalize(pkI, nonce, metadata, blind_sig, blind_inv)
+~~~
+
+The Finalize function is defined in {{PBLINDRSA}}. If this
+succeeds, the Client then constructs a Token as described in [AUTHSCHEME] as
+follows:
+
+~~~
+struct {
+  uint16_t token_type = 0x1234; /* Type Blind RSA (2048-bit) */
+  opaque metadata<1..2^16-1>;
+  uint8_t nonce[32];
+  uint8_t challenge_digest[32];
+  uint8_t token_key_id[32];
+  uint8_t authenticator[Nk];
+} Token;
+~~~
+
+The Token.nonce value is that which was sampled in {{Section 5.1 of PROTOCOL}}.
+If the Finalize function fails, the Client aborts the protocol.
+
+## Token Verification
+
+TODO(shendrick): Replace this entire section with a correct verification description
+
+Verifying a Token requires checking that Token.authenticator is a valid
+signature over the remainder of the token input using the Issuer Public Key.
+The function `RSASSA-PSS-VERIFY` is defined in Section 8.1.2 of !RFC8017}},
+using SHA-384 as the Hash function, MGF1 with SHA-384 as the PSS mask
+generation function (MGF), and a 48-byte salt length (sLen).
+
+~~~
+token_authenticator_input =
+  concat(Token.token_type,
+         Token.nonce,
+         Token.challenge_digest,
+         Token.token_key_id)
+valid = RSASSA-PSS-VERIFY(pkI,
+                          token_authenticator_input,
+                          Token.authenticator)
+~~~
+
+## Issuer Configuration {#public-issuer-configuration}
+
+Issuers are configured with Private and Public Key pairs, each denoted skI and
+pkI, respectively, used to produce tokens. Each key pair SHALL be generated as
+as specified in FIPS 186-4 {{?DSS=DOI.10.6028/NIST.FIPS.186-4}}, where n is 4096 bits in length. These key
+pairs MUST NOT be reused in other protocols.
 
-# Conventions and Definitions
+The key identifier for a keypair (skI, pkI), denoted `token_key_id`, is
+computed as SHA256(encoded_key), where encoded_key is a DER-encoded
+SubjectPublicKeyInfo (SPKI) object carrying pkI. The SPKI object MUST use the
+RSASSA-PSS OID {{!RFC5756}}, which specifies the hash algorithm and salt size.
+The salt size MUST match the output size of the hash function associated with
+the public key and token type.
 
-{::boilerplate bcp14-tagged}
+Since Clients truncate `token_key_id` in each `TokenRequest`, Issuers should
+ensure that the truncated form of new key IDs do not collide with other
+truncated key IDs in rotation.
 
 
```

Comment:
Will review more after our summit, but do we need to add a new token type definition, to modify the table from the protocol document?